### PR TITLE
link test plan items

### PR DESCRIPTION
**Pull request by phoracek**
Signed-off-by: Petr Horacek <phoracek@redhat.com>

**What this PR does / why we need it**:

In order to be able to map Red Hat internal test plan with tests in this
project, we are adding annotations to some tests.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/nnce_conditions_test.go [hunk 1]
```diff
@@ -113,29 +114,47 @@ var _ = Describe("EnactmentCondition", func() {
...
+			wg.Add(len(nodes))
+			for i, _ := range nodes {
+				node := nodes[i]
+				go func() {
+					defer wg.Done()
+					By(fmt.Sprintf("Check %s failing state is kept", node))
+					enactmentConditionsStatusConsistently(node).Should(ConsistOf(failingEnactmentConditions...))
+				}()
+			}
+			wg.Wait()
```
Comment (qinqon) [inline test/e2e/nnce_conditions_test.go]: We do like that for performance reasons ?

Comment (phoracek) [reply test/e2e/nnce_conditions_test.go]: `Consistently` is blocking. If we want to ensure the status is not changing on all the nodes, we have to do it in parallel. Also, it saves some seconds in the test run.

Comment (qinqon) [reply test/e2e/nnce_conditions_test.go]: /lgtm
/approve

Fair enough

**Comment by phoracek** [comment]
@qinqon are those failures caused by this PR or are we experiencing them on master?

**Comment by yossisegev** [comment]
LGTM

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -74,9 +75,15 @@ func TestE2E(tapi *testing.T) {
...
-	knmstateReporter := knmstatereporter.New("test_logs/e2e", framework.Global.Namespace, nodes)
-	junitReporter := ginkgoreporters.NewJUnitReporter("junit.functest.xml")
-	RunSpecsWithDefaultAndCustomReporters(t, "E2E Test Suite", []Reporter{junitReporter, knmstateReporter})
+
+	reporters := make([]Reporter, 0)
+	reporters = append(reporters, knmstatereporter.New("test_logs/e2e", framework.Global.Namespace, nodes))
+	reporters = append(reporters, ginkgoreporters.NewJUnitReporter("junit.functest.xml"))
+	if qereporters.JunitOutput != "" {
+		reporters = append(reporters, qereporters.NewJunitReporter())
+	}
```
Comment (qinqon) [inline test/e2e/main_test.go]: Why this ?

Comment (phoracek) [reply test/e2e/main_test.go]: It is a special version of ginkgo reported maintained by QE. I guess it parses polarion IDs, integrates with BugZilla etc.

**Comment by yossisegev** [comment]
LGTM

**Comment by qinqon** [comment]
/lgtm
/approve
/hold

Let's check that artifacts are correct.

**Comment by qinqon** [comment]
@phoracek artifacts not working propertly

**Pull request closed without merging**